### ISSUE TITLE
hypdestopt: no longer works with current latex-dev

### ISSUE THREAD

**Issue by schlcht** [ho-tex/oberdiek#85]
With current latex-dev (2020.9.1), the following: 
```
\documentclass{article}
\usepackage{hyperref}
\usepackage{hypdestopt}
\begin{document}
\section{A}
\end{document}
```

will never get rid of the warning: 

```
pdfTeX warning (dest): name{section.1} has been referenced but does not exist
, replaced by a fixed one
```

and the bookmark won't work.

Everything's fine with non-dev latex and also was with the previous latex-dev version, so I suppose this has to do with the new hook management.

**Comment by u-fischer** [comment]
yes, the package patches \@begindocumenthook and must be adapted. But I'm a bit surprised that someone actually still uses it and so remarks that there is a problem ;-).
